spec/models/league_spec.rb

```diff
@@ -140,6 +141,146 @@
+        teams = FactoryBot.create_list(:team, 2)
+
+        expect(league.have_already_played(teams[0], teams[1])).to be false
+        expect(league.have_already_played(teams[1], teams[0])).to be false
+
+        league.matches << Match.new(team_home: teams[0], team_away: teams[1], gameday_number: 0)
+
+        expect(league.have_already_played(teams[0], teams[1])).to be true
+        expect(league.have_already_played(teams[1], teams[0])).to be true
+      end
+
+      context 'update schedule' do
+        it 'creates a new gameday' do
+          gameday_amount = league.gamedays.length
+          league.update_schedule
+          expect(league.gamedays.length).to be gameday_amount + 1
+        end
+
+
+        it 'creates a correct amount of matches' do
+          matches_amount = league.matches.length
+          league.update_schedule
+          expect(league.matches.length).to be matches_amount + (league.teams.length / 2).floor
+        end
+
+        it 'does not create matches of teams that already played against each other' do
+          league.update_schedule
+          current_pairings = league.matches.map { |match| Set[match.team_home, match.team_away] }
+          expect(current_pairings.uniq.length == current_pairings.length).to be true
+        end
+
+        it 'creates the matches correctly by the ranking' do
+          league = FactoryBot.create(:league)
+          league.game_mode = League.game_modes[:swiss]
+          teams = FactoryBot.create_list(:team, 6)
+          league.teams.append teams
+          league.gamedays << FactoryBot.create(:gameday)
+
+          match1 = FactoryBot.create(:match, team_home: teams[0], team_away: teams[1], gameday_number: 1,
+                                      points_home: 3, points_away: 0)
+          match1.game_results << FactoryBot.build(:game_result,
+                                                  score_home: 20, # team 0
+                                                  score_away: 0   # team 1
+          )
+          league.matches << match1
+
+          match2 = FactoryBot.create(:match, team_home: teams[2], team_away: teams[3], gameday_number: 1,
+                                     points_home: 3, points_away: 0)
+          match2.game_results << FactoryBot.build(:game_result,
+                                                  score_home: 18, # team 2
+                                                  score_away: 4   # team 3
+          )
+          league.matches << match2
+
+          match3 = FactoryBot.create(:match, team_home: teams[4], team_away: teams[5], gameday_number: 1,
+                                     points_home: 3, points_away: 0)
+          match3.game_results << FactoryBot.build(:game_result,
+                                                  score_home: 15, # team 4
+                                                  score_away: 3   # team 5
+          )
+          league.matches << match3
```

Comment:
at least extract all this code into an extra method inside the test file so we have a method name that says what it does please :) and maybe put this in a let statement because it has nothing to do with what you are testing at all.

Reply:
Does not seem much prettier :D
What do you want to extract? Creating all the matches?

Reply:
yes, inside the it statement should actually only be the code that you are testing. You aren't really testing how match creation works or if you can add it to the matches of leagues, so it should be in a before block or rather a lazy evaluated let.

Also extracting this into a method enabled you to use it later or use it again in a test with a different ranking if you parametrize the method smarty.